app/vmselect/graphite/transform.go

```diff
@@ -3599,6 +3599,17 @@ func groupSeriesByNodes(ss []*series, nodes []graphiteql.Expr) map[string][]*ser
 	return m
 }
 
+func getAbsoluteNodeIndex(index, size int) (int, error) {
+	// handling the negative index case
+	if index < 0 && index+size > 0 {
+		index = index + size
+	}
+	if index >= size || index < 0 {
+		return index, fmt.Errorf("index out of bound")
```

Comment:
I think, it's better to return -1 in case, when index out of bound. It's not a good practice to ignore error from return, cause it may confuse future readers. 

```go
func getAbsoluteNodeIndex(index, size int) int {
	// handling the negative index case
	if index < 0 && index+size > 0 {
		index = index + size
	}
	if index >= size || index < 0 {
		return -1
        }
        return index
```

and use it:
```go
if n := getAbsoluteNodeIndex(t.N,len(parts)); n >= 0 && n < len(parts) {
				dstParts = append(dstParts, parts[n])
			}
```

 cc @valyala

Reply:
I did not test again that **n < len(parts)** a it's checked in the function :smile:. I usually tend to prefer using an explicit error in stead of a special value, but in this case I agree that it makes more sense